pandas/core/common.py

```diff
@@ -225,14 +225,27 @@ def count_not_none(*args) -> int:
     return sum(x is not None for x in args)
 
 
-def asarray_tuplesafe(values, dtype: NpDtype | None = None) -> np.ndarray:
+@overload
+def asarray_tuplesafe(
+    values: ArrayLike | list | tuple | zip, dtype: NpDtype | None = ...
+) -> np.ndarray:
+    # ExtensionArray can only be returned when values is an Index, all other iterables
+    # will return np.ndarray. Unfortunately "all other" cannot be encoded in a type
+    # signature, so instead we special-case some common types.
```

Comment:
Not sure whether it would reduce ignore comments but you could add a second overload with Index -> ExtensionArray

Reply:
The returned type when passing an index is `ArrayLike`, not necessarily `ExtensionArray`. You could overload for specific Index subtypes, but that currently would not remove any ignore comments.